koswat/cost_report/infrastructure/multi_infrastructure_profile_costs_calculator_builder.py

```diff
@@ -0,0 +1,102 @@
+import math
+from dataclasses import dataclass
+
+from koswat.configuration.settings.costs.infastructure_costs_settings import (
+    InfrastructureCostsSettings,
+)
+from koswat.configuration.settings.costs.surtax_costs_settings import (
+    SurtaxCostsSettings,
+)
+from koswat.configuration.settings.koswat_general_settings import SurtaxFactorEnum
+from koswat.core.protocols.builder_protocol import BuilderProtocol
+from koswat.cost_report.infrastructure.infrastructure_costs_calculator import (
+    InfrastructureCostsCalculator,
+)
+from koswat.cost_report.infrastructure.multi_infrastructure_profile_costs_calculator import (
+    MultiInfrastructureProfileCostsCalculator,
+)
+from koswat.dike.surroundings.wrapper.infrastructure_surroundings_wrapper import (
+    InfrastructureSurroundingsWrapper,
+)
+
+
+@dataclass
+class MultiInfrastructureProfileCostsCalculatorBuilder(BuilderProtocol):
+
+    infrastructure_wrapper: InfrastructureSurroundingsWrapper
+    cost_settings: InfrastructureCostsSettings
+
+    # We are only interested in the `roads_` properties.
+    surtax_cost_settings: SurtaxCostsSettings
+
+    def _get_surtax_costs(self) -> float:
+        if not self.surtax_cost_settings:
+            return math.nan
+
+        if self.infrastructure_wrapper.storage_cost_factor == SurtaxFactorEnum.NORMAAL:
+            return self.surtax_cost_settings.roads_normal
+
+        if self.infrastructure_wrapper.storage_cost_factor == SurtaxFactorEnum.MOEILIJK:
+            return self.surtax_cost_settings.roads_hard
+
+        return self.surtax_cost_settings.roads_easy
+
+    def _get_infrastructure_costs(
+        self, infrastructure_name: str
```

Comment:
I do not think so, technically they are yet another "surrounding", which are properties of the infrastructure / obstacle wrappers.

This one in particular has been a pain, because I do not necessarily need to give knowledge onto which infrastructure is, it makes my life easier later on during the costs calculation as it avoids the need to access the class properties' names with `getattr` (I did this as a first implementation, but I found it very flaw).